### ISSUE TITLE
Y4M Input doesn't work anymore and Issues with YUV of long duration

### ISSUE THREAD

**Issue by Karam150** [fraunhoferhhi/vvenc#228]
Since Vvenc 1.6.1 Y4M Input doesn't work anymore, it's the same thing with the 1.7 RC1.  I have this message "Parameter Check Error : TicksPerSecond should be a multiple of FrameRate/Framescale"

There is an issue with YUV file of long duration, I tested a file of 3 minutes and the video appears with bug (like I put wrong parameters (resolution and fps) in command line. But disappeared with the same video but cutted (30 seconds).

**Comment by jbrdbg** [comment]
Which FrameRate has your sequence? Could you try to set TicksPerSecond = 2700000 on the command line? E.g. for expert app: 
`vvencFFapp --TicksPerSecond 2700000`
or for easy app:
`vvencapp --tickspersec 2700000`

For your second report, could you provide us with the command line you are using for the encodings or which would be helpful with an example for us to check?

**Comment by Karam150** [comment]
For Y4M input the framerate of sequence was 24 fps (23.98 fps), the command line is : vvencapp.exe --preset medium -i Alliance01.y4m -b 2000000 -o Alliance01Medium2000Kbpsv170RC1.266, but it works with this option "TicksPerSecond = 2400000".

I made a test with the version 1.6, and it works without adding "TicksPerSecond = 2400000"

![Vvenc 160](https://user-images.githubusercontent.com/120000030/206314260-e4b0a052-a06d-4ba8-b24c-1f161947a87e.jpg)

![Vvenc 170RC1](https://user-images.githubusercontent.com/120000030/206314354-cb00e437-5715-4d2b-b7ba-bf2a7968c018.jpg)


For the second report, the command line is : vvencapp.exe --preset medium -i Alliance01.yuv -s 1280x544 -r 24 -b 2000000  -o Alliance01Medium2000Kbpsv170.266

**Comment by lehmann-c** [comment]
Concerning the TicksPerSecond there was an issue in version <= 1.6 when generating timestamps.
As 90000 is the default for TicksPerSecond, NTSC framerates needs a higher resolution as 90000 for TicksPerSecond.
Therefore there is an error now. But we will adapt this behaviour.

Concerning the 2nd issue. Can you give more details what happens? 
How or with what player do you play the .266 file? Do you play .266 or the decoded file?
Could it be a playback issue?
Maybe the decoder you are using is to slow to playback in realtime and starts to stutter?

**Comment by Karam150** [comment]
I decode the -.266 files with vvdecapp : vvdecapp.exe -b Alliance01Medium2000Kpsv170.266 -o Alliance01Medium2000Kpsv170.y4m

I play the decoded file. 

The video looks like this : 

![Alliance](https://user-images.githubusercontent.com/120000030/206578780-fc6a3652-0ca0-45cc-879a-0d131fe5e0ce.jpg)

**Comment by jbrdbg** [comment]
To me this looks, like there is a small shift in the input YUV, which then sums up over a long sequence and leads to visible artefacts. Is your input video plain yuv or y4m? Based on your command line (Alliance01.yuv) it should be yuv, but if this file is an y4m, this would explain the small shift, due to the y4m header data in the file.
One way to check, would be to dump the first bytes of your video. If it starts with "FRAME", then this is a y4m video and no plain yuv.
Alternatively you could force the encoder to read the input as y4m, by adding --y4m to your command line (--y4m [0] force y4m input (only needed for input pipe, else enabled by .y4m file extension). In case the input is no y4m the encoder will detect this at the beginning and stop encoding, otherwise the encoder should start encoding your video.

**Comment by Karam150** [comment]
Thanks for the fix .

After research , it's not a real YUV , but a Y4M file. I took a VC-1 file with AVS script and I converted it with ffmpeg.

The command line : ffmpeg -i Alliance.avs -pix_fmt yuv420p -f yuv4mpegpipe Alliance.yuv.

But the output is a "yuv4mpegpipe". (See the screenshot below)

![Alliance yuv](https://user-images.githubusercontent.com/120000030/206975550-d2f1b1de-f2eb-4151-9fef-361fa0ac2df4.jpg)


Thanks for comment.
